### PR TITLE
Add types for locomotive-scroll

### PR DESCRIPTION
**Pull request by colinlienard**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by peterblazejewicz** [comment]
Hey,
this looks like UMD module, allowing global script:
https://github.com/locomotivemtl/locomotive-scroll#or-without
https://github.com/locomotivemtl/locomotive-scroll/blob/master/dist/locomotive-scroll.js#L2-L7
https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html#identifying-a-umd-library

**Comment by ColinLienard** [comment]
Hello,
Do I need to do this ?
```ts
declare global {
  // All of my types
}
```
I not really familiar with this syntax.

**Comment by gabritto** [comment]
> 

I think you actually need a line saying `export as namespace LocomotiveScroll;`. However, I see that the original package also has an esm entry point, so I think you'll also need to configure that. I'll come back with instructions on how to do it soon 😅

**Review by gabritto — CHANGES_REQUESTED** [review]
See my comment about `export as namespace` :)

**Review by gabritto — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
this can be also written (supporting CommonJS in umd):
```ts
// Type definitions for locomotive-scroll 4.1
// Project: https://github.com/locomotivemtl/locomotive-scroll
// Definitions by: Colin Lienard <https://github.com/ColinLienard>
// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped

declare namespace LocomotiveScroll {
    type Direction = 'vertical' | 'horizontal';

    type Offset = [number | string, number | string];

    ...
    ...

    interface ScrollToOptions {
        offset?: number;
        callback?: () => void;
        duration?: number;
        easing?: [number, number, number, number];
        disableLerp?: boolean;
    }
}
declare class LocomotiveScroll {
    constructor(options?: LocomotiveScroll.InstanceOptions);
    init(): () => void;
    on<EventName extends 'scroll' | 'call'>(
        eventName: EventName,
        callback: (event: EventName extends 'scroll' ? LocomotiveScroll.OnScrollEvent : string | string[]) => void,
    ): () => void;
    update(): () => void;
    destroy(): () => void;
    start(): () => void;
    stop(): () => void;
    scrollTo(target: LocomotiveScroll.ScrollToTarget, options?: LocomotiveScroll.ScrollToOptions): () => void;
}

export as namespace LocomotiveScroll;

export = LocomotiveScroll;
```

File: types/locomotive-scroll/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        "noImplicitAny": true,
+        "noImplicitThis": true,
+        "strictFunctionTypes": true,
+        "strictNullChecks": true,
+        "baseUrl": "../",
+        "typeRoots": ["../"],
+        "types": [],
+        "noEmit": true,
+        "forceConsistentCasingInFileNames": true,
+        "esModuleInterop": true
```
Comment (gabritto) [inline types/locomotive-scroll/tsconfig.json:14]: Do you need to enable this option? We usually don't want DT packages to enable extra options, and leave things like `esModuleInterop` to be used by consumers of the DT definitions.

**Review by gabritto — CHANGES_REQUESTED** [review]

File: types/locomotive-scroll/tsconfig.json (hunk 1 above)
Comment (ColinLienard) [reply types/locomotive-scroll/tsconfig.json:14]: If I don't set `esModuleInterop` to `true`, then in my test file I get this error in VSCode: *This module is declared with using 'export =', and can only be used with a default import when using the 'esModuleInterop' flag.* and the typing doesn't work.

Also I get *SyntaxError: Unexpected token } in JSON at position 390* when running `npm test locomotive-scroll` with `esModuleInterope` set to `false`.

I might just go back to where I was before the support for CommonJS. What do you think is the best ?

File: types/locomotive-scroll/locomotive-scroll-tests.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import LocomotiveScroll from 'locomotive-scroll';
```
Comment (peterblazejewicz) [inline types/locomotive-scroll/locomotive-scroll-tests.ts:1]:
```suggestion
import LocomotiveScroll = require('locomotive-scroll');
```
https://github.com/DefinitelyTyped/DefinitelyTyped#esmoduleinteropallowsyntheticdefaultimports

File: types/locomotive-scroll/tsconfig.json (hunk 1 above)
Comment (peterblazejewicz) [inline types/locomotive-scroll/tsconfig.json:14]:
```suggestion
        "forceConsistentCasingInFileNames": true
```
see comment on text, about `esModuleInterop`, thanks!

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@ColinLienard congrats on your first contribution here!

**Comment by ColinLienard** [comment]
Ready to merge

**Pull request merged**